src/utils/math/zoom.js

```diff
@@ -10,19 +10,26 @@ function transform(scale, start, end, factor) {
   return [newStart, newEnd];
 }
 
-export default function zoom({ e, chart, pointComponent, viewHandler, pinchZoomFactor }) {
+export default function zoom({ e, chart, pointComponent, viewHandler, pinchZoomFactor, buttonZoomDirection }) {
   const { scale, maxScale, minScale } = viewHandler.getMeta();
-  const zoomFactor = pinchZoomFactor || (e.deltaY > 0 ? ZOOM_SCALE : 1 / ZOOM_SCALE);
+  const { xAxisMin, xAxisMax, yAxisMin, yAxisMax } = viewHandler.getDataView();
+  let zoomFactor;
+  let fixedPoint;
+  if (buttonZoomDirection) {
+    zoomFactor = buttonZoomDirection === 'out' ? ZOOM_SCALE : 1 / ZOOM_SCALE;
+    fixedPoint = { normX: 0.5, normY: 0.5 };
+  } else {
+    zoomFactor = pinchZoomFactor || (e.deltaY > 0 ? ZOOM_SCALE : 1 / ZOOM_SCALE);
+    const p = eventToComponentPoint(e, chart, pointComponent);
+    const { width, height } = pointComponent.rect.computedPhysical;
+    fixedPoint = { normX: p.x / width, normY: p.y / height };
+  }
   const newScale = zoomFactor * scale;
-
   if (newScale > maxScale || newScale < minScale) {
     return;
   }
-  const p = eventToComponentPoint(e, chart, pointComponent);
-  const { width, height } = pointComponent.rect.computedPhysical;
-  const { xAxisMin, xAxisMax, yAxisMin, yAxisMax } = viewHandler.getDataView();
-  const [xMin, xMax] = transform(p.x / width, xAxisMin, xAxisMax, zoomFactor);
-  const [yMax, yMin] = transform(p.y / height, yAxisMax, yAxisMin, zoomFactor);
+  const [xMin, xMax] = transform(fixedPoint.normX, xAxisMin, xAxisMax, zoomFactor);
+  const [yMax, yMin] = transform(fixedPoint.normY, yAxisMax, yAxisMin, zoomFactor);
   viewHandler.setDataView({
```

Comment:
Actually the same goes for this function, it's performing zoom (updating the dataView of the viewHandler). I know you haven't added this file now, but it's not a typical "util" file either, so would be nice to move it somewhere else, like into the viewHandler

Reply:
good idea!

Reply:
fixed!